addons/mail/static/src/models/file_uploader.js

```diff
@@ -70,16 +70,24 @@ registerModel({
          * @param {Object} param0.attachmentData
          * @param {Composer} param0.composer
          * @param {Thread} param0.thread
+         * @param {boolean} param0.fromDraft
          * @returns {Attachment}
          */
-        _onAttachmentUploaded({ attachmentData, composer, thread }) {
+        _onAttachmentUploaded({ attachmentData, composer, thread, fromDraft = false } = {}) {
             if (attachmentData.error || !attachmentData.id) {
                 this.messaging.notify({
                     type: 'danger',
                     message: attachmentData.error,
                 });
                 return;
             }
+            if (fromDraft) {
+                return {
+                    composer: composer,
+                    originThread: (!composer && thread) ? thread : undefined,
+                    ...attachmentData,
+                };
+            }
```

Comment:
Could `const data = {}`, so that the branching is returning `data` or `Attachment.insert(data)`.

Reply:
Done